File: 7_kyu/Simple_string_characters.py
from typing import List

def solve(s: str) -> List[int]:
    uppercase = 0
    lowercase = 0
    numbers = 0
    special = 0
    for c in s:
        if c.isdigit():
            numbers += 1
        elif c.isupper():
            uppercase += 1
        elif c.islower():
            lowercase += 1
        else:
            special += 1
    return [uppercase,lowercase,numbers,special]

File: 7_kyu/Simple_string_characters.go
package kata

import "unicode"

func Solve(s string) []int {
	cnt_upper := 0
	cnt_lower := 0
	cnt_digit := 0
	cnt_other := 0
	for _, c := range s {
		switch {
		case unicode.IsUpper(c):
			cnt_upper++
		case unicode.IsLower(c):
			cnt_lower++
		case unicode.IsDigit(c):
			cnt_digit++
		default:
			cnt_other++
		}
	}
	return []int{cnt_upper, cnt_lower, cnt_digit, cnt_other}
}
